syntax = "proto3";

package anya;

service AnyaIM {
    rpc Convert(ConvReq) returns (ConvRsp) {}
    rpc PartialConvert(PartConvReq) returns (PartConvRsp) {}
}

message ConvReq {
    string in_str = 1;
}

message ConvRsp {
    int32 status = 1;
    string out_str = 2;
}

message PartConvReq {
    string in_str = 1;
}

message PartConvRsp {
    int32 status = 1;
    repeated string out_str = 2;
}
